paasta_tools/api/client.py

```diff
@@ -65,10 +65,39 @@ def get_paasta_api_client(
         spec_dict = json.load(f)
     # replace localhost in swagger.json with actual api server
     spec_dict['host'] = api_server
+    spec_dict['schemes'] = [parsed.scheme]
 
     # sometimes we want the status code
+    requests_client = PaastaRequestsClient(
+        scheme=parsed.scheme,
+        cluster=cluster,
+        system_paasta_config=system_paasta_config,
+    )
     if http_res:
         config = {'also_return_response': True}
-        return SwaggerClient.from_spec(spec_dict=spec_dict, config=config)
+        return SwaggerClient.from_spec(spec_dict=spec_dict, config=config, http_client=requests_client)
     else:
-        return SwaggerClient.from_spec(spec_dict=spec_dict)
+        return SwaggerClient.from_spec(spec_dict=spec_dict, http_client=requests_client)
+
+
+class PaastaRequestsClient(RequestsClient):
+    def __init__(
+        self,
+        scheme: str,
+        cluster: str,
+        system_paasta_config: SystemPaastaConfig,
+    ):
+        if scheme == 'https':
+            if system_paasta_config.get_enable_client_cert_auth():
+                home = os.environ.get('HOME', '')
+                ssl_cert = (f'{home}/paasta.crt', f'{home}/paasta.key')
```

Comment:
cert/key should be different for different clusters, right?
Also, we probably better put them into a subdirectory inside `$HOME`, something like `$HOME/.paasta/pki/mesosstage.crt`?

Reply:
👍good point. I nearly made this configurable and was just being lazy because I didn't want to solve the $HOME substitution side. Let me take another look.

Reply:
You can use something like
```
dir = os.path.expanduser('~/.paasta/pki')
ssl_cert = (f'{dir}/{cluster}.crt', f'{dir}/{cluster}.key')
```

Reply:
Nice, don't think I knew about `expanduser`!